### PR TITLE
tstest/integration: exercise TCP DNS queries against quad-100

### PR DESCRIPTION
**Pull request by jwhited**
Updates tailscale/corp#22511

### CODE REVIEW

**Review by bradfitz — APPROVED** [review]

**Review by danderson — APPROVED** [review]
Thank you for writing the regression test!

File: tstest/integration/integration_test.go [hunk 1]
```diff
@@ -1118,6 +1120,107 @@ func TestAutoUpdateDefaults(t *testing.T) {
...
+	wantIP4 := n1.AwaitIP4()
+	wantIP6 := n1.AwaitIP6()
+	n1.AwaitRunning()
+
+	status, err := n1.Status()
+	if err != nil {
+		t.Fatalf("failed to get node status: %v", err)
+	}
+	selfFQDN := status.Self.DNSName
+	if !strings.HasSuffix(selfFQDN, ".") {
```
[the remark this answers is not in the document]
Comment (jwhited) [reply tstest/integration/integration_test.go]: updated in https://github.com/tailscale/tailscale/pull/13231/commits/72e65b641ffcab4f8a07b32e7a598e1be9f1f419

**Comment by jwhited** [comment]
If I run this with a high `count=n` locally I sometimes see this:

```
    integration_test.go:1204: got (143.244.220.150) != want (100.64.0.1)
```

printing the DNS response shows:

```
    integration_test.go:1202: resp: ;; opcode: QUERY, status: NOERROR, id: 63790
        ;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 0

        ;; QUESTION SECTION:
        ;primary.	IN	 A

        ;; ANSWER SECTION:
        primary.	46	IN	A	143.244.220.150
```

Either the quad-100 resolver is acting recursively, sometimes? Or, the dns pkg client is doing some fallback I don't yet understand. 143.244.220.150 is some AT&T DNS error address IIUC, or just some bad leak somewhere.

I don't want to merge a flake or a test that false positives, so looking closer.

**Comment by jwhited** [comment]
The test is racey around routing table changes. Sometimes the DNS queries end up flowing over a default route to another host running tailscaled, explaining the wildcard AT&T error response addr. I probably want to add some sort of await to prevent this race.

**Comment by jwhited** [comment]
> The test is racey around routing table changes. Sometimes the DNS queries end up flowing over a default route to another host running tailscaled, explaining the wildcard AT&T error response addr. I probably want to add some sort of await to prevent this race.

With the test checking the DNS response I chose not to add any routing table checks. I also removed `t.Paralell()` to prevent DNS requests from leaking across test nodes.

https://github.com/tailscale/tailscale/commit/641693d61cc6b4dbe8b2767ca23e887d43c0ba0d added the IPv6 service addr route, which fixed the remaining issue with IPv6 failing.

**Pull request merged**